assets/src/edit-story/components/header/title.js

```diff
@@ -58,7 +64,19 @@ function Title() {
       actions: { updateStory },
     }) => ({ title, slug, updateStory })
   );
-  const { setTitleInput } = useHeader();
+  const { setTitleInput, titleInput } = useHeader();
+  const { onFocusOut, highlight } = useHighlights((state) => ({
+    highlight: state[states.STORY_TITLE],
+    onFocusOut: state.onFocusOut,
+  }));
+
+  useEffect(() => {
+    if (highlight?.focus) {
+      titleInput?.focus();
+    }
```

Comment:
💭 this may be a silly recommendation, but we could do this on one line
```suggestion
    highlight?.focus && titleInput?.focus();
```